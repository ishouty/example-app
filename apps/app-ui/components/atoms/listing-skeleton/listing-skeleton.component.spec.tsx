import { render, screen } from '@testing-library/react';
import ListingSkeleton from './listing-skeleton.component';

describe('ListingSkeleton Component', () => {
  test('matches snapshot', () => {
    const { container } = render(<ListingSkeleton />);

    expect(container).toMatchSnapshot();
  });
  test('renders 5 skeleton items', () => {
    const { container } = render(<ListingSkeleton />);

    // Check if the skeleton avatar renders 5 times
    const skeletonAvatars = container.querySelectorAll('.listitem');
    expect(skeletonAvatars).toHaveLength(5);
  });

  test('each skeleton has an avatar and two text lines', () => {
    render(<ListingSkeleton />);
    const listItem = screen.getAllByTestId('listitem');
    expect(listItem.length).toBe(5);

    listItem.map((item) => {
      expect(item.getElementsByClassName('line').length).toBe(2);
    });
  });
});
