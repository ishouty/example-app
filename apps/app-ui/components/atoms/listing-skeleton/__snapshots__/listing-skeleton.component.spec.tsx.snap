// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingSkeleton Component matches snapshot 1`] = `
<div>
  <div
    class="space-y-4"
  >
    <div
      class="flex items-center p-4 bg-white shadow rounded-lg listitem"
      data-testid="listitem"
    >
      <div
        class="h-12 w-12 bg-gray-200 rounded-full animate-pulse mr-4 avatar"
      />
      <div
        class="flex-1 space-y-2"
      >
        <div
          class="h-4 bg-gray-200 rounded w-3/4 animate-pulse line"
        />
        <div
          class="h-4 bg-gray-200 rounded w-5/6 animate-pulse line"
        />
      </div>
    </div>
    <div
      class="flex items-center p-4 bg-white shadow rounded-lg listitem"
      data-testid="listitem"
    >
      <div
        class="h-12 w-12 bg-gray-200 rounded-full animate-pulse mr-4 avatar"
      />
      <div
        class="flex-1 space-y-2"
      >
        <div
          class="h-4 bg-gray-200 rounded w-3/4 animate-pulse line"
        />
        <div
          class="h-4 bg-gray-200 rounded w-5/6 animate-pulse line"
        />
      </div>
    </div>
    <div
      class="flex items-center p-4 bg-white shadow rounded-lg listitem"
      data-testid="listitem"
    >
      <div
        class="h-12 w-12 bg-gray-200 rounded-full animate-pulse mr-4 avatar"
      />
      <div
        class="flex-1 space-y-2"
      >
        <div
          class="h-4 bg-gray-200 rounded w-3/4 animate-pulse line"
        />
        <div
          class="h-4 bg-gray-200 rounded w-5/6 animate-pulse line"
        />
      </div>
    </div>
    <div
      class="flex items-center p-4 bg-white shadow rounded-lg listitem"
      data-testid="listitem"
    >
      <div
        class="h-12 w-12 bg-gray-200 rounded-full animate-pulse mr-4 avatar"
      />
      <div
        class="flex-1 space-y-2"
      >
        <div
          class="h-4 bg-gray-200 rounded w-3/4 animate-pulse line"
        />
        <div
          class="h-4 bg-gray-200 rounded w-5/6 animate-pulse line"
        />
      </div>
    </div>
    <div
      class="flex items-center p-4 bg-white shadow rounded-lg listitem"
      data-testid="listitem"
    >
      <div
        class="h-12 w-12 bg-gray-200 rounded-full animate-pulse mr-4 avatar"
      />
      <div
        class="flex-1 space-y-2"
      >
        <div
          class="h-4 bg-gray-200 rounded w-3/4 animate-pulse line"
        />
        <div
          class="h-4 bg-gray-200 rounded w-5/6 animate-pulse line"
        />
      </div>
    </div>
  </div>
</div>
`;
